libcaf_net/caf/net/web_socket/framing.cpp

```diff
@@ -10,6 +10,49 @@
 
 namespace caf::net::web_socket {
 
+// -- static utility functions -------------------------------------------------
+
+expected<std::pair<uint16_t, const_byte_span>>
+framing::decode_closing_payload(const_byte_span payload) {
+  if (payload.size() == 0)
+    return std::make_pair(0, payload);
+  if (payload.size() < 2)
+    return make_error(caf::sec::protocol_error,
+                      "no status flag in closing payload");
+  auto status = (std::to_integer<uint_fast16_t>(payload[0]) << 8)
+                + std::to_integer<uint16_t>(payload[1]);
+  if (!detail::rfc3629::valid(payload.subspan(2)))
+    return make_error(sec::protocol_error,
+                      "malformed text message in closing payload");
+  if (status < 1000 || status >= 5000)
+    return make_error(sec::protocol_error,
+                      "invalid status code in closing payload");
```

Comment:
I believe this is redundant. The `if`-block below pass for anything in range [3000, 5000). Then, we call `from_integer`, which converts known values to an `enum` value. This will always fail for any `x < 1000` and any `x > 5000` by definition.

```suggestion
```